pulse_lib/virtual_channel_constructors.py

```diff
@@ -20,28 +30,59 @@ def __init__(self, pulse_lib_obj, name=None):
         self.virtual_gate_names = []
         self._virtual_gate_matrix = None
         self.valid_indices = None
+        self.square = square
+        if inverse:
+            if square:
+                raise ValueError('Cannot have inverse and square true.')
+            self.vgm = self._fetch_virtual_gate_matrix
+            self.vgmi = self._inv_virtual_gate_matrix
+        else:
+            self.vgm = self._inv_virtual_gate_matrix
+            self.vgmi = self._fetch_virtual_gate_matrix
 
     @property
     def virtual_gate_matrix(self):
+        # Property for backwards compatibilty
+        return self.vgm()
+    
+    @property
+    def virtual_gate_matrix_inv(self):
+        # Property for backwards compatibilty
+        return self.vgmi()
+    
+    def _fetch_virtual_gate_matrix(self):
+        '''
+        returns the virtual gate matrix from the np object. checks which of the
+        gates exist in the pulselib and limits the matrix to these gates.
+        '''
         if self._virtual_gate_matrix is None:
-            raise ValueError("Cannot fetch virutal gate matrix, please define (see docs).")
+            raise ValueError("Cannot fetch virtual gate matrix, please define (see docs).")
 
         self.virtual_gate_matrix_tmp = np.asarray(self._virtual_gate_matrix)
         self.virtual_gate_matrix_tmp = self.virtual_gate_matrix_tmp[self.valid_indices]
-        self.virtual_gate_matrix_tmp = self.virtual_gate_matrix_tmp[:,self.valid_indices]
+        if self.square:
+            self.virtual_gate_matrix_tmp = self.virtual_gate_matrix_tmp[:,self.valid_indices]
```

Comment:
I guess the same condition applies to the selection of the valid virtual gates in load_via_hardware